### PR TITLE
Blog App - processing data in models

### PR DESCRIPTION
**Pull request by zewdu444**
### Blog App - processing data in models

- [x] Set up associations between models.
- [x] Write SQL queries with Active Record.
- [x] Custom methods were added for each models
- [x] All Methods tested in` rails console`
- [x] Linter has been checked

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -1,5 +1,13 @@
 class Post < ApplicationRecord
-  belongs_to :user, foreign_key: :author_id, class_name: 'User', counter_cache: :posts_counter
+  belongs_to :author, foreign_key: :author_id, class_name: 'User', counter_cache: :posts_counter
   has_many :comments, dependent: :destroy
   has_many :likes, dependent: :destroy
+
+  def update_user_post_counter
+    author.update(posts_counter: author.posts.count)
+  end
```
Comment (OLIPLICHE) [inline app/models/post.rb]: # Optional
- [ ] Hi, you have done a great Job. I would suggest you use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html); that is `after_save` to call the `#update_post_counter` method every time a new post has been saved to a database, then make the `update_post_counter` method private: Example
```ruby
  after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end
```

File: app/models/comment.rb [hunk 2]
```diff
@@ -1,4 +1,8 @@
 class Comment < ApplicationRecord
-  belongs_to :user, foreign_key: :author_id, class_name: 'User'
+  belongs_to :author, foreign_key: :author_id, class_name: 'User'
   belongs_to :post, class_name: 'Post', counter_cache: :comments_counter
+
+  def update_post_comment_counter
+    post.update(comments_counter: post.comments.count)
+  end
```
Comment (OLIPLICHE) [inline app/models/comment.rb]: # Optional: 
- Can you use the same suggestion I made in models/comments: `after_save :update_comments_counter`

File: app/models/user.rb [hunk 3]
```diff
@@ -1,5 +1,9 @@
...
-  has_many :posts ,dependent: :destroy
-  has_many :comments, dependent: :destroy
-  has_many :likes, dependent: :destroy
+  has_many :posts, foreign_key: :author_id, dependent: :destroy
+  has_many :comments, foreign_key: :author_id, dependent: :destroy
+  has_many :likes, foreign_key: :author_id, dependent: :destroy
+
+  def most_three_recent_post
+    posts.order(created_at: :desc).limit(3)
+  end
```
Comment (OLIPLICHE) [inline app/models/user.rb:8]: # Optional:
- Can you use the same suggestion I made in models/comments: `after_save :update_likes_counter`

**Review by OLIPLICHE — APPROVED** [review]
Hi @zewdu444, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Highlights
- Created associations between models👏
- Created custom methods in `User`, `post`, `comments` and `likes`

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by zewdu444** [comment]
@OLIPLICHE  thank you for valuable feedback I updated accordingly

**Pull request merged**